File: fetch_images.js
const fs = require('fs');
const https = require('https');

const queries = [
    { name: 'islemci.png', q: 'AMD RYZEN 5 7600X' },
    { name: 'ekran-karti.png', q: 'ASUS PRIME RTX 5070 WHITE' },
    { name: 'anakart.png', q: 'MSI B850M GAMING PLUS WHITE' },
    { name: 'ram.png', q: 'LEXAR ARES 16 GB 6400 MHz WHITE RAM' },
    { name: 'sogutucu.png', q: 'ASUS MAX Gaming LC 360mm ARGB LCD White' },
    { name: 'kasa.png', q: 'BITFENIX MH100 WHITE pc case' },
    { name: 'ssd.png', q: 'TEAM T-FORCE 1TB SSD' },
    { name: 'fan.png', q: 'LIAN LI UNI FAN CL Wireless White' },
    { name: 'monitor1.png', q: 'AOC 310 Hz 0.3 MS FAST IPS monitor' },
    { name: 'monitor2.png', q: 'Casper Excalibur 200 Hz 1ms FAST IPS' },
    { name: 'mouse.png', q: 'LOGITECH G PRO X SUPERLIGHT 2 WHITE' },
    { name: 'klavye.png', q: 'LOGITECH G515 TKL WHITE' },
    { name: 'kulaklik.png', q: 'LOGITECH G733 WHITE' },
    { name: 'mousepad.png', q: 'Wraith Spirit of Aim Pro Hybrid' },
    { name: 'skatez.png', q: 'Hoverpad V3' },
    { name: 'koltuk.png', q: 'Hawk Fab c5 White Fabric gaming chair' }
];

if (!fs.existsSync('ekipmanlar')) fs.mkdirSync('ekipmanlar');

async function download() {
    for (const item of queries) {
        console.log(`Searching: ${item.q}`);
        const url = `https://html.duckduckgo.com/html/?q=${encodeURIComponent(item.q + " product image")}`;
        
        await new Promise(resolve => {
            https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
                let data = '';
                res.on('data', chunk => data += chunk);
                res.on('end', () => {
                    const match = data.match(/src="\/\/external-content\.duckduckgo\.com\/iu\/\?u=([^&"]+)/);
                    if (match && match[1]) {
                        const imgUrl = decodeURIComponent(match[1]);
                        console.log(`Found: ${imgUrl}`);
                        https.get(imgUrl, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (imgRes) => {
                            if (imgRes.statusCode === 200) {
                                const file = fs.createWriteStream(`ekipmanlar/${item.name}`);
                                imgRes.pipe(file);
                                file.on('finish', () => {
                                    console.log(`Saved ${item.name}`);
                                    file.close();
                                    resolve();
                                });
                            } else {
                                console.log(`Failed to download image ${imgRes.statusCode}`);
                                resolve();
                            }
                        }).on('error', resolve);
                    } else {
                        console.log('No image found');
                        resolve();
                    }
                });
            }).on('error', resolve);
        });
        
        // Wait 2 sec
        await new Promise(r => setTimeout(r, 2000));
    }
}

download();

File: fetch_images.py
import os
import re
import urllib.request
import urllib.parse
import time
from PIL import Image

equipment = [
    ("islemci.png", "AMD RYZEN 5 7600X box"),
    ("ekran-karti.png", "ASUS PRIME RTX 5070 WHITE"),
    ("anakart.png", "MSI B850M GAMING PLUS WHITE"),
    ("ram.png", "LEXAR ARES 16 GB 6400 MHz WHITE RAM"),
    ("sogutucu.png", "ASUS MAX Gaming LC 360mm ARGB LCD White"),
    ("kasa.png", "BITFENIX MH100 WHITE"),
    ("ssd.png", "TEAM T-FORCE 1TB SSD"),
    ("fan.png", "LIAN LI UNI FAN CL Wireless White"),
    ("monitor1.png", "AOC 310 Hz 0.3 MS FAST IPS monitor"),
    ("monitor2.png", "Casper Excalibur 200 Hz 1ms FAST IPS"),
    ("mouse.png", "LOGITECH G PRO X SUPERLIGHT 2 WHITE"),
    ("klavye.png", "LOGITECH G515 TKL WHITE"),
    ("kulaklik.png", "LOGITECH G733 WHITE"),
    ("mousepad.png", "Wraith Spirit of Aim Pro Hybrid"),
    ("skatez.png", "Hoverpad V3"),
    ("koltuk.png", "Hawk Fab c5 White Fabric")
]

os.makedirs("ekipmanlar", exist_ok=True)

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
}

for filename, query in equipment:
    filepath = os.path.join("ekipmanlar", filename)
    if os.path.exists(filepath):
        print(f"{filename} already exists.")
        continue

    print(f"Searching for {query}...")
    url = "https://www.bing.com/images/search?q=" + urllib.parse.quote(query)
    
    try:
        req = urllib.request.Request(url, headers=headers)
        with urllib.request.urlopen(req) as response:
            html = response.read().decode('utf-8')
        
        # Parse Bing image search JSON data
        matches = re.findall(r'murl&quot;:&quot;(http[^&]+)&quot;', html)
        if not matches:
             matches = re.findall(r'murl":"([^"]+)"', html)

        if matches:
            img_url = matches[0]
            print(f"Found image: {img_url}")
            
            req_img = urllib.request.Request(img_url, headers=headers)
            with urllib.request.urlopen(req_img, timeout=10) as res, open(filepath, 'wb') as f:
                f.write(res.read())
            
            # Resize image
            try:
                img = Image.open(filepath).convert("RGBA")
                img.thumbnail((400, 400))
                img.save(filepath, "PNG")
                print(f"Saved {filename}")
            except Exception as e:
                print(f"Failed to resize {filename}: {e}")
        else:
            print("No matches found.")
    except Exception as e:
        print(f"Error fetching {query}: {e}")
    time.sleep(1)

print("Done!")
